agent/server.go

```diff
@@ -147,26 +178,37 @@ func newRPCServer(agent *agent, caFile, certFile, keyFile string, rpcAddrs []str
 
 	// handler is a multiplexer for both gRPC and HTTPS queries.
 	// The HTTPS endpoint returns the cluster status as JSON
+	// TODO: why does server need to handle both gRPC and HTTPS queries?
```

Comment:
When do we use HTTPS instead of gRPC?

Reply:
We use HTTPS to support a simple curl-style query for status.